scripts/deploy.sh

```diff
@@ -1,88 +1,47 @@
-    echo $GCLOUD_SERVICE_KEY | base64 --decode > ${HOME}/gcloud-service-key.json
-    gcloud auth activate-service-account --key-file ${HOME}/gcloud-service-key.json
-}
-
-configure_google_cloud_sdk() {
-    echo "Configuring Google Cloud Sdk"
-    gcloud auth activate-service-account --key-file=${HOME}/gcloud-service-key.json
-    gcloud --quiet config set project ${GOOGLE_PROJECT_ID}
-    gcloud --quiet config set compute/zone ${GOOGLE_COMPUTE_ZONE}
-    gcloud --quiet container clusters get-credentials ${GOOGLE_CLUSTER_NAME}
-}
-
-login_to_docker_registry() {
-    echo "====> Login to docker registry"
-    docker login -u _json_key --password-stdin https://${DOCKER_REGISTRY} < ${HOME}/gcloud-service-key.json
-}
+source $DEPLOY_SCRIPT_PATH
 
-logout_of_docker_registry() {
-    echo "====> Logout from docker"
-    docker logout https://${DOCKER_REGISTRY}
-}
-
-tag_and_publish_image() {
-    docker build -t $IMAGE -f docker/release/Dockerfile .
-    docker push $IMAGE
-}
-
-deploy_to_kubernetes() {
-    # TODO: deploy built image to kubernetes cluster
-    # gcr.io/bench-projects/wire-backend:
-    echo "====> Prepare image for deployement"
-    echo "====> Deploying ${IMAGE} to ${DEPLOYMENT_NAME} in ${ENVIRONMENT} environment"
-    kubectl set image deployment/${DEPLOYMENT_NAME} backend=${IMAGE} -n ${ENVIRONMENT}
-
-    if [ "$?" == "0" ]; then
-        echo "Deployment completed succesfully"
-        exit 0
-    else
-        echo "Failed to deploy ${IMAGE} to ${ENVIRONMENT} environment"
-        exit 1
-    fi
-}
-
-main () {
-    install_google_cloud_sdk
-    auth_with_service_account
-    configure_google_cloud_sdk
-    login_to_docker_registry
-    tag_and_publish_image
-    logout_of_docker_registry
-    deploy_to_kubernetes
-}
-
-main
+DOCKER_REGISTRY=gcr.io
+GCLOUD_SERVICE_KEY_NAME=gcloud-service-key.json
+ALLOWED_DEPLOY_ENVIRONMENTS=('staging', 'production')
+
+require 'PRODUCTION_GOOGLE_COMPUTE_ZONE' $PRODUCTION_GOOGLE_COMPUTE_ZONE
+require 'STAGING_GOOGLE_COMPUTE_ZONE' $STAGING_GOOGLE_COMPUTE_ZONE
+require 'STAGING_CLUSTER_NAME' $STAGING_CLUSTER_NAME
+require 'PRODUCTION_CLUSTER_NAME' $PRODUCTION_CLUSTER_NAME
+require 'PROJECT_NAME' $PROJECT_NAME
+require 'GOOGLE_PROJECT_ID' $GOOGLE_PROJECT_ID
+require 'DOCKER_REGISTRY' $DOCKER_REGISTRY
+require 'GCLOUD_SERVICE_KEY' $GCLOUD_SERVICE_KEY
+
+BRANCH_NAME=$CIRCLE_BRANCH
+# set deployment environment
+setEnvironment $BRANCH_NAME
+# ensure its an allowed deployment environment
+isAllowedDeployEnvironment $ENVIRONMENT
+# get K8s deployment name
+
+getDeploymentName DEPLOYMENT_NAME
+
+# Set image image tag and name
+IMAGE_TAG=$(getImageTag $(getCommitHash))
+IMAGE_NAME=$(getImageName)
+
+main() {
+    installGoogleCloudSdk
+    authWithServiceAccount
+    configureGoogleCloudSdk
+    loginToContainerRegistry _json_key
+    buildAndTagDockerImage .
+    publishDockerImage
+    logoutContainerRegistry $DOCKER_REGISTRY
+    deployToKubernetesCluster bot
```

Comment:
this should be `deployToKubernetesCluster backend` since the container name is named backend not bot.